worker/uniter/runner/context/context.go

```diff
@@ -532,16 +535,13 @@ func (ctx *HookContext) SetPodSpec(specYaml string) error {
 		return ErrIsNotLeader
 	}
 	entityName = ctx.unit.ApplicationName()
-	err = ctx.state.SetPodSpec(entityName, specYaml)
+	dryRun := true
```

Comment:
I am not sure we need dry run. It can be implemented slightly differently.
The use of dry-run basically gets us the parsing of the spec (and hence any surfacing of errors) without actually setting it on the model yet. This parsing check could be done here in the hook context. Currently ParsePodSpec() is a method on the k8s environ provider but there's no reason the method can't be changed to be a package method and called here. Doing it this different way would negate the need for a facade version bump and greatly simplify the PR.